github.go

```diff
@@ -201,48 +220,92 @@ func unauthenticatedGet(pages int, page int, client *http.Client) (GithubEvents,
 	}
-	r, err := httpClient.Get("https://api.github.com/events?page=" + strconv.Itoa(page))
+	r, err := httpClient.Get(url)
 	if err != nil {
 		log.Fatalf("Error in requesting data from API: %s", err.Error())
 	}
-	defer r.Body.Close()
 
-	RateLimitSpecs.Limit = parseHeader(r.Header, "x-ratelimit-limit")
-	RateLimitSpecs.Remaining = parseHeader(r.Header, "x-ratelimit-remaining")
-	RateLimitSpecs.ResetTimestamp = parseLongHeader(r.Header, "x-ratelimit-reset")
-	RateLimitSpecs.PollInterval = parseHeader(r.Header, "x-poll-interval") * pages
+	defer r.Body.Close()
 
 	if r.StatusCode == http.StatusNotModified { // (304) No new content
 		return nil, &APIError{"no new content", 304}
 	} else if r.StatusCode == http.StatusForbidden { // (403) Rate limit reached
 		return nil, &APIError{"rate limited", 403}
 	}
 
-	body, _ := ioutil.ReadAll(r.Body)
-	var events GithubEvents
-	json.Unmarshal(body, &events)
+	body, err := ioutil.ReadAll(r.Body)
 
-	return events, nil
+	if err != nil {
+		log.Fatal("Unable to read response body")
+		return nil, err
+	}
+
+	return body, nil
 }
 
 // authenticatedGet performs an authenticated call to the GitHub's API using the user-supplied token.
 // A custom call to `unauthenticatedGet` is made under the hood.
-func authenticatedGet(pages int, page int, token string) (GithubEvents, error) {
+func authenticatedGet(url string, token string) ([]byte, error) {
 	ts := oauth2.StaticTokenSource(
 		&oauth2.Token{AccessToken: token},
 	)
 	tc := oauth2.NewClient(oauth2.NoContext, ts)
 
-	return unauthenticatedGet(pages, page, tc)
+	return unauthenticatedGet(url, tc)
 }
 
 // GetHubData returns a success/failure boolean (respectively `true`/`false`) along with the marshalled API data.
 // See the `GithubEvents` struct.
 func GetHubData(pages int, page int, token string) (GithubEvents, error) {
-	if token != "" {
-		return authenticatedGet(pages, page, token)
+	var responseBody []byte
+	var err error
+
+	if token == "" {
+		responseBody, err =
+			unauthenticatedGet("https://api.github.com/events?page="+strconv.Itoa(page), nil)
+	} else {
+		responseBody, err =
+			authenticatedGet("https://api.github.com/events?page="+strconv.Itoa(page), token)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	var events GithubEvents
+	json.Unmarshal(responseBody, &events)
+
+	return events, nil
+}
+
+// GetRateLimits fetches rate limits from GitHub API server for the current client. It returns a `rateLimitSpecs`
+// data structures containg reset timestamp, used requests, max requests and a default polling interval
+func GetRateLimits(token string) rateLimitSpecs {
+	// TODO: Check if rateLimitSpecs is already initialized
+
+	var response []byte
+	var err error
+
+	const rateLimitURL = "https://api.github.com/rate_limit"
+	if token == "" {
+		response, err = unauthenticatedGet(rateLimitURL, nil)
+	} else {
+		response, err = authenticatedGet(rateLimitURL, token)
+	}
+
+	if err != nil {
+		panic("Error while fetching rate limits for GitHub API")
```

Comment:
Is not being able to fetch RL an unrecoverable error? I'd say not, because the default delay fallback should kick in: `panic`king will quit the program instead.